components/scream/src/physics/common/physics_saturation_impl.hpp

```diff
@@ -10,13 +10,76 @@ namespace physics {
  * Implementation of saturation functions. Clients should NOT #include
  * this file, #include physics_functions.hpp instead.
  */
+
+template <typename S, typename D>
+KOKKOS_FUNCTION
+void  Functions<S,D>
+::check_temp(const Spack& temp, const std::string& fname)
+{
+  Smask is_neg_temp = (temp <= 0);
+  if (is_neg_temp.any()){
+    scream::error::runtime_abort("Error: Called from:"+ fname +"; Temperature has <= 0 values");
+  }
+  /*
+  Smask is_nan_temp = isnan(temp);
+  if (is_nan_temp.any()){
+    scream::error::runtime_abort("Error: Called from:"+ fname +"; Temperature has NaN values");
+  }
+
+  Smask is_finite_temp = isfinite(temp);
+  if (is_finite_temp.any()){
+      scream::error::runtime_abort("Error: Called from:"+ fname +";  Temperature has infinite values");
+    }*/
```

Comment:
why are NaN and isfinite checks commented out?

Reply:
`isnan`was not implemented for packs but now it is (thanks to @bartgol), so I will activate this code now. `isfinite` is still missing and I will see if I can follow what Luca did for `isnan` to implement `isfinite`.

Reply:
I tried activating it but the C++/Kokkos version fails complaining that there are NaNs in the input. It most likely seems to be due to the use of packs. 
@jgfouca and @bartgol : Do you expect the tests to pass if we are checking NaNs in temperature pack?

Reply:
@singhbalwinder - this comment is resolved now, right?

Reply:
@singhbalwinder 

I assume the NaNs you don't want to check are in inactive pack indices? If that's the case, you can pass a mask to limit the check.